pkg/reconcile/pipeline/infinispan/handler/manage/hotrod_upgrades.go

```diff
@@ -469,17 +465,19 @@ func (r *HotRodRollingUpgradeRequest) syncData() error {
 		return fmt.Errorf("error obtaining the existing statefulSet '%s': %w", targetStatefulSetName, err)
 	}
 
-	podList, err := r.PodsCreatedBy(targetStatefulSetName)
+	podListTarget, err := r.PodsCreatedBy(targetStatefulSetName)
 	if err != nil {
 		return fmt.Errorf("failed to obtain pods from the target cluster: %w", err)
 	}
-
-	sourceClient, err := ctx.InfinispanClient()
+	podListSource, err := r.PodsCreatedBy(getSourceStatefulSetName(r.i))
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to obtain pods from the source cluster: %w", err)
 	}
+
 	// Clone the source curl client as the credentials are the same, updating the pod to one from the target statefulset
-	targetClient := ctx.InfinispanClientForPod(podList.Items[0].Name)
+	targetClient := ctx.InfinispanClientForPod(podListTarget.Items[0].Name)
+	sourceClient := ctx.InfinispanClientForPod(podListSource.Items[0].Name)
```

Comment:
This change still doesn't take into account the operand version of the source cluster, as the code for this method is:

```go
	curlClient := c.curlClient(podName)
	return ispnClient.New(c.Operand(), curlClient)
```

Where `c.Operand` returns the `spec.version`.

I still think we need what I describe [here](https://github.com/infinispan/infinispan-operator/pull/2370#discussion_r2445396599).